lib/ASTDecl.cpp

```diff
@@ -60,27 +63,43 @@ namespace pillar
         assert(body.getBlocks().empty());
         return func_decl;
       }
+      AST::lift_queue->emplace_back([&body, func_decl, functionName, this](void)
+                                    {
+                                      // Lift each statement from the function body, collecting them into
+                                      // `body_stmts`.
+                                      std::vector<clang::Stmt *> body_stmts;
+                                      for (mlir::Operation &op : body.front())
+                                      {
 
-      // Lift each statement from the function body, collecting them into
-      // `body_stmts`.
-      std::vector<clang::Stmt *> body_stmts;
-      for (mlir::Operation &op : body.front())
-      {
-        if (clang::Stmt *stmt = LiftOp(func_decl, op))
-        {
-          if (!ElideFromCompoundStmt(op, stmt))
-          {
-            body_stmts.emplace_back(stmt);
-          }
-        }
-      }
+                                        if (clang::Stmt *stmt = LiftOp(func_decl, op))
+                                        {
+                                          if (!ElideFromCompoundStmt(op, stmt))
+                                          {
+                                            body_stmts.emplace_back(stmt);
+                                          }
+                                        }
+                                      }
+
+                                      clang::FPOptionsOverride fpo;
+                                      clang::CompoundStmt *body_stmt = clang::CompoundStmt::Create(
+                                          ctx, body_stmts, fpo, clang::SourceLocation(), clang::SourceLocation());
```

Comment:
I think there's a `kEmptyLoc` that you can use in place of `clang::SourceLocation()`. You should see it used elsewhere.